frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+const { utils } = require( "ethers" );
+import { delayMsec } from "../utils/utils";
+
+import Typography from "@mui/material/Typography";
+import {
+	Button,
+	Container,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import Grid from "@mui/material/Grid";
+import ConditionalAlert from "./ConditionalAlert";
+
+function AccountEntry({ data }) {
+	let balance = "NaN";
+	// data.balance can have NaN value if no Metamask or other Wallet installed
+	if (!isNaN(data.balance)) {
+		balance = utils.formatEther(data.balance);
+	}
+	
+	return <TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{balance}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthereumAccountDetails ({ publicAddresses, setErrorNoMetamaskMessage }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		// TODO: publicAddresses is still not loaded. 
+		// Solve temporarily by waiting for 0.5 sec.
+		await delayMsec(500);
+		const provider = await detectEthereumProvider();
+		let details = [];
+		if (!provider) {
+			setErrorNoMetamaskMessage("Couldn't get data from Ethereum Provider. Please install MetaMask!");
+			console.error("no ethereum provider");
+			// Display only ETH address from backend
+			publicAddresses.map(pubAddress => {
+				details.push({ "pubAddress": pubAddress, "balance": NaN });
+			}) ;
+			setDetailData(details);
+			return;
+		}
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				details.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(details);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
```

Comment:
I think, this should be `ParagraphTypography` because of design. It would be quite low effort to change it in this PR compared to someone changing this later in an upcoming PR (where git-diffs get larger).

Therefore, I'd like to have `ParagraphTypography` instead of `Typography`.